arcaflow_plugin_horreum_client/horreum_client_schema.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+import enum
+import typing
+import re
+from dataclasses import dataclass
+from arcaflow_plugin_sdk import schema
+
+
+class access(str, enum.Enum):
+    PUBLIC = "PUBLIC"
+    PROTECTED = "PROTECTED"
+    PRIVATE = "PRIVATE"
+
+
+@dataclass
+class InputParams:
+    horreum_url: typing.Annotated[
+        str,
+        schema.name("Horreum URL"),
+        schema.description(
+            "The complete base URL for the Horreum server,"
+            " such as 'https://horreum.example.com'."
+        ),
+        schema.pattern(
+            re.compile(
+                r"(https?:\/\/)?"
+                r"[a-zA-Z0-9_-]{1,63}(\.[a-zA-Z0-9_-]{1,63})*"
+                r"(:[0-9]{1,5})?"
+                r"\/?"
+            )
+        ),
+    ]
+    horreum_keycloak_url: typing.Annotated[
+        str,
+        schema.name("Horreum Keycloak URL"),
+        schema.description(
+            "The complete base URL for the Horreum Keycloak server,"
+            " such as 'https://keycloak.example.com'."
+        ),
+        schema.pattern(
+            re.compile(
+                r"(https?:\/\/)?"
+                r"[a-zA-Z0-9_-]{1,63}(\.[a-zA-Z0-9_-]{1,63})*"
+                r"(:[0-9]{1,5})?"
+                r"\/?"
+            )
+        ),
+    ]
+    horreum_username: typing.Annotated[
+        str,
+        schema.name("Horreum username"),
+        schema.description("Username for the Horreum server."),
+    ]
+    horreum_password: typing.Annotated[
+        str,
+        schema.name("Horreum password"),
+        schema.description("Password for the Horreum server."),
+    ]
+    test_name: typing.Annotated[
+        str,
+        schema.name("test name"),
+        schema.description(
+            "Name of the target test in Horreum for the object being uploaded."
+        ),
+    ]
+    test_owner: typing.Annotated[
+        str,
+        schema.name("test owner"),
+        schema.description("Owner of the object being uploaded."),
+    ]
+    test_access_rights: typing.Annotated[
+        access,
+        schema.name("test access rights"),
+        schema.description("Access rights for the object being uploaded."),
+    ]
```

Comment:
Isn't "access" basically an ENUM with something like private/protected/public? Can/should that be checked here?

Reply:
Yes -- The enum is defined above in the schema and referenced here with `access` (the enum name) as the data type.

https://github.com/arcalot/arcaflow-plugin-horreum-client/blob/07eb4ba2e79d9f24d1a0b1b46998a8efbeda3fe7/arcaflow_plugin_horreum_client/horreum_client_schema.py#L10-L13

Reply:
Indeed. I'll blame my brain (it wasn't *my* fault) for reading the annotation here too quickly!